### PR TITLE
support cert name for generic mdm data provider

### PR DESCRIPTION
**Pull request by Sneezry**

### CODE REVIEW

File: src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs [hunk 1]
```diff
@@ -32,4 +35,64 @@ public void PostInitialize()
...
+        /// </summary>
+        public void PostInitialize()
+        {
+        }
+
+        public GenericMdmDataProviderConfigurationWrapper(GenericMdmDataProviderConfiguration config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("Configuration is null.");
```
Comment (nmallick1) [inline src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs]:
```suggestion
                throw new ArgumentNullException("Supplied MDM configuration is null.");
```

File: src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs [hunk 2]
```diff
@@ -32,4 +35,64 @@ public void PostInitialize()
...
+            }
+            if (config.MonitoringAccount != null && config.MonitoringAccount.StartsWith("Mock"))
+            {
+                CertificateThumbprint = "BAD0BAD0BAD0BAD0BAD0BAD0BAD0BAD0BAD0BAD0";
+            }
+            else
+            {
+                using (X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine))
+                using (X509Store userStore = new X509Store(StoreName.My, StoreLocation.CurrentUser))
```
Comment (nmallick1) [inline src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs]: In the master branch, the cert is expected to be present only in the Computer store. No need to iterate through the user store. Iterating certificate store is an expensive operation..

File: src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs [hunk 3]
```diff
@@ -32,4 +35,64 @@ public void PostInitialize()
...
+                    store.Open(OpenFlags.ReadOnly);
+                    userStore.Open(OpenFlags.ReadOnly);
+
+                    var certificateCollection = store.Certificates;
+                    certificateCollection.AddRange(userStore.Certificates);
+                    var certificates = certificateCollection.Find(X509FindType.FindBySubjectName, config.CertificateName, true);
+
+                    if (certificates.Count == 0)
+                    {
+                        throw new SystemException("Certificate is not found.");
```
Comment (nmallick1) [inline src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs]:
```suggestion
                        throw new SystemException($"MDM certificate with subject name {config.CertificateName} not found in Computer store.");

```

File: src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs [hunk 4]
```diff
@@ -32,4 +35,64 @@ public void PostInitialize()
...
+            {
+                using (X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine))
+                using (X509Store userStore = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+                {
+                    store.Open(OpenFlags.ReadOnly);
+                    userStore.Open(OpenFlags.ReadOnly);
+
+                    var certificateCollection = store.Certificates;
+                    certificateCollection.AddRange(userStore.Certificates);
+                    var certificates = certificateCollection.Find(X509FindType.FindBySubjectName, config.CertificateName, true);
```
Comment (nmallick1) [inline src/Diagnostics.DataProviders/DataProviderConfigurations/GenericMdmDataProviderConfiguration.cs]:
```suggestion
                    var certificates = store.Certificates.Find(X509FindType.FindBySubjectName, config.CertificateName, true);
```

**Comment by Sneezry** [comment]
All done. Thanks for the suggestions!

**Pull request merged**